src/waveform/renderers/qopengl/shaders/gradientshader.cpp

```diff
@@ -0,0 +1,28 @@
+#include "waveform/renderers/qopengl/shaders/gradientshader.h"
+
+using namespace qopengl;
+
+void GradientShader::init() {
+    QString vertexShaderCode = QStringLiteral(R"--(
+uniform mat4 matrix;
+attribute vec4 position;
+attribute vec3 gradient;
+varying vec3 vGradient;
+void main()
+{
+    vGradient = gradient;
+    gl_Position = matrix * position;
+}
+)--");
+
+    QString fragmentShaderCode = QStringLiteral(R"--(
+uniform vec4 color;
+varying vec3 vGradient;
+void main()
+{
+    gl_FragColor = vec4(color.x, color.y, color.z, color.w * max(0.0, abs((vGradient.x + vGradient.y) * 4.0 - 2.0) - 1.0));
```

Comment:
Looking at this again I feel it is a) overly complicated and b) misleading because the name suggests this is something generic while it reality its for drawing a very specific gradient (which fades from color.w to transparent in the first quarter of either x or y axis, and back to color.w in the last quarter).

I will either rename this (and explain with a comment) or do the same by using more generic code.

Reply:
I have reimplemented this with a much simpler shader (RGBAShader) that interpolates between to color vertices to draw the same gradients.